thrift/compiler/main.h

```diff
@@ -24,6 +24,8 @@
 #include "thrift/compiler/parse/t_const.h"
 #include "thrift/compiler/parse/t_field.h"
 
+extern "C" {
```

Comment:
@kapcino thanks for the feedback.

When initially patching the file, I considered adding the guards in the event of a C compiler being used over a file including `compiler/main.h`. However, I decided against it because within FBThrift, the files using this header would never be successfully built with a C compiler. It's also important to note that [other parts of the header do not work without the C++ STL](https://github.com/micahsaint/fbthrift/blob/flexfix/thrift/compiler/main.h#L77-L87). By wrapping the code in the guards, we imply to the code reader that there may be a time where a C compiler is used over files including this header and currently, as far as I can tell, that's never the case.